### ISSUE TITLE
[New Item] Unpaintable Cube

### ISSUE THREAD

**Issue by Luke18033** [TeamSpen210/BEE2-items#585]
An item which is identical to the normal Cube item, but the cubes cannot be painted. If this isn't possible, paint simply wouldn't affect the cubes (kind of like frankenturrets). There could also be an alternate unpaintable version of any future custom cubes which are added.

**Comment by TeamSpen210** [comment]
I could parent a paint fizzler to the cube, which should instantly un-paint it when painted. I might use the Contraption Cube model ir something for this. It'll just be the standard cube, not all the varieties.

**Comment by Luke18033** [comment]
Why can't you use the normal cube dropper properties menu (with cube type, auto drop etc.) to allow switching between cubes?

**Comment by TeamSpen210** [comment]
I could actually, but I'd prefer to make it look like a new cube entirely. Otherwise it might be confusing when mixed with regular cubes, since you can't tell it apart otherwise.

**Comment by Luke18033** [comment]
I got this idea when playing an Aperture Tag map I made. It involves covering a cube in repulsion gel at one point, and I thought, "How are people supposed to know that you're supposed to cover the cube in repulsion gel?" I thought that if there was an unpaintable cube, people would know that they needed to cover the cube in paint because the unpaintable cube wasn't being used. It could also prevent cheating or getting stuck in some of my other (unpublished) chambers.

**Comment by TeamSpen210** [comment]
Maybe I should add the 'cube shower' signage type to the Signage item.

**Comment by Luke18033** [comment]
@TeamSpen210 Wait, what does signage have to do with unpaintable cubes?

**Comment by TeamSpen210** [comment]
That sign might help with indicating that the cube should be painted.

**Comment by Luke18033** [comment]
> It could also prevent cheating or getting stuck in some of my other (unpublished) chambers.

Wait, you're not saying that your going to implement the signage of the unpaintable cube, are you?

**Comment by Luke18033** [comment]
A "no-paint" icon could be put in the following places on the cubes:

- [ ] Standard/Companion \- The aperture logo/heart would be replaced on some sides.
- [ ] Reflection \- It would be a translucent tint on the glass parts.
- [ ] Sphere \-  It would be at the places where the three lines meet.
- [ ] Frankenturrets \- On the back or top, there will be the sign. Not sure how useful this would be though, because frankenturrets aren't affected by paint anyway (Valve-bug, I'm assuming?)

**Comment by TheDarkBomber** [comment]
@Luke18033

Actually, Frankenturrets aren't cubes, so they can't be painted anyway. A frakenturret is actually a form of turret. (Frankenturret not prop_weighted_cube)

**Comment by TeamSpen210** [comment]
Turrets can be painted too... Frankenturrets are their own entity, `prop_monster_box`. It's indeed a Valve bug.

**Comment by TheDarkBomber** [comment]
No, it is because Frankenturrets move. Turrets do not move.

**Comment by ZombiePwner1** [comment]
You can make it so when the cube is painted it gets fizzled and another one respawns.

**Comment by Luke18033** [comment]
@ZombiePwner1 No, that's a bad idea. As TeamSpen210 said, there should be a paint fizzler parented to the cube which would make it instantly un-paint.

**Comment by ZombiePwner1** [comment]
I don,t know how that's gonna work but give it a shot

**Comment by Luke18033** [comment]
@TeamSpen210 How about just have paint fizzler effects surrounding the cube?

**Comment by Luke18033** [comment]
@TheDarkBomber Frankenturrets being unpaintable *is* a bug. See [here](https://developer.valvesoftware.com/wiki/Prop_monster_box).

**Comment by TheDarkBomber** [comment]
@Luke18033 How do you know it's not a fake source?

**Comment by Luke18033** [comment]
It's the Valve Developer Community. When have they been wrong?

**Comment by TheDarkBomber** [comment]
When vandalism goo + momentum = tragic comedy.

That statement was vandalised.....

**Comment by Luke18033** [comment]
Looking back at this now, I think it would be a good idea to make this a separate cube type entirely, like TeamSpen originally suggested. I'm not sure what it should look like, probably some sort of grating/frame cube, maybe similar to the Contraption Cube.
